Source/WebKit/UIProcess/API/Cocoa/WKWebView.h

```diff
@@ -666,6 +666,26 @@ The uniform type identifier kUTTypeWebArchive can be used get the related pasteb
  */
 @property (nonatomic, readonly, getter=isWritingToolsActive) BOOL writingToolsActive WK_API_AVAILABLE(macos(15.0), ios(18.0), visionos(WK_XROS_TBA));
 
+/* @enum WKWebViewDataType
+   @abstract The type of WKWebView data.
+   @constant WKWebViewDataTypeSessionStorage Session Storage data.
+*/
+typedef NS_OPTIONS(NSUInteger, WKWebViewDataType) {
+    WKWebViewDataTypeSessionStorage = 1 << 0
+} WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA));
+
+/* @abstract Called when the client wants to fetch WKWebView data.
+   @param dataTypes The option set of WKWebView data types whose data the client wants to fetch.
+   @param completionHandler The completion handler that should be invoked with the retrieved data and possibly an error. The retrieved data will be a serialized blob. If an error occurred, the retrieved data will be nil. An error may occur if a requested data type is not supported or if the data cannot be retrieved for some other reason (such as a crash).
```

Comment:
How should we test for a user requesting an unsupported data type? Is it even possible? They have to use the NS_OPTIONS above.